TestProject/src/rules/BaseRule.php

```diff
@@ -0,0 +1,39 @@
+<?php
+namespace GameOfLife\Rules;
+
+use GameOfLife\options\Getopt;
+use GameOfLife\cellularAutomat\Field;
+
+/**
+ * API-class for additional rules.
+ */
+abstract class BaseRule
+{
+    /**
+     * @param Field $_field The cell which is going to be checked.
+     * @return bool The new state of the given field in the next generation
+     */
+    public function calculateNewState(Field $_field)
+    {
+    }
+
+    /**
+     * Add rule specific-options to the option list.
+     *
+     * @param Getopt $_options option list.
+     * @return void
+     */
+    public function addOptions(Getopt $_options)
+    {
+    }
+
+    /**
+     * Initializes the rule.
+     *
```

Comment:
Please explain when this is called and what it can be used for, so that the developer implementing a rule knows what to do with it!